src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1493,6 +1500,19 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
         except Exception as ex:
             tr_logger.error("Error occured while checking if the MSI certificate has expired: {}".format(str(ex)), exc_info=True)
 
+        collect_logs(cmd, client, resource_group_name, cluster_name, storage_account, sas_token, 
```

Comment:
Have you tested these functions/command on your end? If not, I can help you testing these.

Reply:
Yes, this az connectedk8s troubleshoot command has been tested with the addition of collect_logs. But feel free to test on your end as well.